src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,56 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+/**
+ * Represents a Person's contact in the address book.
+ * Guarantees: immutable; 
+ */
+public class Contact {
+	public static final String CONTACT_VALIDATION_REGEX = ".+";
+
+	public final String value;
+	private boolean isPrivate;
+
+	/**
+	 * Validates given contact.
+	 *
+	 * @throws IllegalValueException if given contact string is invalid.
+	 */
+	public Contact(String contact, boolean isPrivate, String constraints) throws IllegalValueException {
+		this.isPrivate = isPrivate;
+		contact = contact.trim();
+		if (!isValidContact(contact)) {
+			throw new IllegalValueException(constraints);
+		}
+		this.value = contact;
+	}
+
+	/**
+	 * Checks if a given string is a valid person contact.
+	 */
+	public boolean isValidContact(String test) {
+		return test.matches(CONTACT_VALIDATION_REGEX);
+	}
+	+    @Override
+	public String toString() {
+		return value;
+	}
+
+	@Override
+	public boolean equals(Object other) {
```

Comment:
This implementation will allow you to compare Email and Address objects. Although syntactically correct, it is not semantically correct. You should implement this in each children class separately.